### ISSUE TITLE
Escaped quote in the input text value is handled incorrectly

### ISSUE THREAD

**Issue by niolap** [siodb/siodb#147]
```sql
siocli> insert into TABLE1 values ('\'') ;

Status 2: at (1, 30): no viable alternative at input 'values ('\'''

Elapsed time: 2 ms.
```

**Comment by cxxman** [comment]
I doubt this is valid SQL. If I remember correctly it should be just `'''`.

**Comment by cxxman** [comment]
Some database accept some of such sequences, for example
- [PostgreSQL](https://www.postgresql.org/docs/12/sql-syntax-lexical.html#SQL-SYNTAX-STRINGS), see section 4.1.2.2
- [MySQL](https://dev.mysql.com/doc/refman/8.0/en/string-literals.html), see table 9.1

Seems like Siodb SQL parser doesn's support this, but this is not a bug in sense of SQL standard.

**Comment by cxxman** [comment]
But normally in SQL you must write something like:
```sql
'This is one line' + chr(13) + 'This is another line'
```
But we still yet don't support this.

**Comment by niolap** [comment]
Indeed, I created this issue as a placeholder. But after further investigation, the double single quote can be used as a standard and is yet handled in the parser. However, the final value kept by the IOMgr doesn't get rid of the extra quote:

````sql
siocli> insert into d.t values ( '''' ) ;
1 rows affected
Elapsed time: 40 ms.

siocli> select * from d.t;

TRID                C
------------------- ----------------------------------------
                  1 ''

Row(s): 1

Elapsed time: 41 ms.
```

So, we should remove this extra quote to store properly the user value. This is the purpose of this issue.

**Comment by cxxman** [comment]
closed via #150
